File: src/main/java/nl/rabobank/assessment/ui/rest/model/response/PersonResponse.java
package nl.rabobank.assessment.ui.rest.model.response;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import java.time.LocalDate;
import java.util.UUID;

public record PersonResponse(

		UUID id,

		@NotNull
		String firstName,

		@NotNull
		String lastName,

		@NotNull
		LocalDate dateOfBirth,

		@NotNull
		String street,

		@NotNull
		int houseNumber,

		String houseNumberAdditions,

		@NotNull
		@Pattern(regexp = "(?i)^\\d{4}\\s?[A-Z]{2}$")
		String postalCode,

		@NotNull
		String city,

		@NotNull
		String country
) {}

File: src/main/kotlin/nl/rabobank/assessment/ui/rest/model/response/PersonResponse.kt
package nl.rabobank.assessment.ui.rest.model.response

import java.time.LocalDate
import java.util.*
import javax.validation.constraints.Pattern

data class PersonResponse(
    val id: UUID,
    val firstName: String,
    val lastName: String,
    val dateOfBirth: LocalDate,
    val street: String,
    val houseNumber: Int,
    val houseNumberAdditions: String?,
    @Pattern(regexp = "(?i)^\\d{4}\\s?[A-Z]{2}$")
    val postalCode: String,
    val city: String,
    val country: String
)